airbyte-integrations/connectors/source-mongodb-v3/src/main/kotlin/io/airbyte/integrations/source/mongodb/MongoDbSourceConfiguration.kt

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.mongodb
+
+import com.mongodb.MongoCredential
+import io.airbyte.cdk.ConfigErrorException
+import io.airbyte.cdk.command.CdcSourceConfiguration
+import io.airbyte.cdk.command.SourceConfigurationFactory
+import io.airbyte.cdk.ssh.SshConnectionOptions
+import io.airbyte.cdk.ssh.SshNoTunnelMethod
+import io.airbyte.cdk.ssh.SshTunnelMethodConfiguration
+import jakarta.inject.Singleton
+import java.time.Duration
+
+data class MongoDbSourceConfiguration(
+    override val maxConcurrency: Int,
+    override val resourceAcquisitionHeartbeat: Duration,
+    val connectionString: String,
+    val queueSize: Int,
+    val discoverSampleSize: Int,
+    val captureMode: MongoDbSourceConfigurationFactory.CaptureMode,
+    val mongoCredential: MongoCredential?,
+    val schemaEnforced: Boolean,
+    val initialWaitTime: Duration,
+    val initialLoadTime: Duration,
+    val invalidCdcCursorPositionBehavior:
+        MongoDbSourceConfigurationFactory.InvalidCdcCursorPositionBehavior
+) : CdcSourceConfiguration {
+    override val global: Boolean = true
+
+    // Following SSH configuration does not apply to MongoDB.
```

Comment:
We can't tunnel to MongoDB? Is it because it's a cluster and therefore we can't resolve a hostname? If so, this PR should block on making the appropriate fix to the CDK, which assumes we can always have a tunnel.